python_chess_ai/main.py

```diff
@@ -16,32 +16,47 @@
 
 __version__ = "0.1-Alpha"
 
-# usage argument parser: [-h][-t | -g][-v]
+# usage argument parser: [-h] [-t | -g] [-p PLAYER PLAYER][-pT {User,AI,Player,Dummy} {User,AI,Player,Dummy}][-pD {0,1,2,3} {0,1,2,3}] [-v]
 parser = argparse.ArgumentParser()
 ui_group = parser.add_mutually_exclusive_group()
 ui_group.add_argument("-t", "--terminal", help="starts the terminal ui", action="store_true")
 ui_group.add_argument("-g", "--gui", help="starts the GUI", action="store_true")
+player_name_type_group = parser.add_argument_group()
+player_name_type_group.add_argument("-p", "--player", nargs=2, help=argparse.SUPPRESS) # help="set name player 1 and 2"
+player_name_type_group.add_argument("-pT","--player_type", nargs=2, choices=["User","AI", "Player", "Dummy"], help=argparse.SUPPRESS) # help="enter player type for player 1 and 2"
+player_name_type_group.add_argument("-pD","--player_difficulty", nargs=2, type=int, choices=range(0,4), help=argparse.SUPPRESS) # help="enter ai difficulty for player 1 and 2, use 0 if you are not using ai"
 parser.add_argument("-v", "--version", help="print the version number and exit", action="store_true")
 args = parser.parse_args()
+
 def main():
     if args.gui or (not (args.terminal) and not (args.gui) and not (args.version)):
         ui_status = 1
         print("Start GUI")
-        start_chess_master(ui_status)
     elif args.terminal:
         ui_status = 0
-        start_chess_master(ui_status)
+        if not (args.player is None and args.player_type is None and args.player_difficulty is None):
+            interrogate_settings = []
+            for i in range(len(args.player)):
+                interrogate_setting = {
+                    "player_num": i, "player_name": args.player[i], "player_type": args.player_type[i], "player_difficulty": args.player_difficulty[i]}
+                interrogate_settings.append(interrogate_setting)
+
+            settings_ui = ui_switcher(ui_status).Settings()
+            player_settings = settings_ui.fast_interrogate_settings(interrogate_settings)
+            start_chess_master(ui_status, settings_ui, player_settings)
+        else:
+            settings_ui = ui_switcher(ui_status).Settings()
+            player_settings = settings_ui.interrogate_settings()
+            start_chess_master(ui_status, settings_ui, player_settings)
```

Comment:
würde das ganze anders machen, hier ist mir wieder zu viel verzweigt und so. 

Übergebe an interrogate_settings immer die player arguments

Bei interrogate_settings fügst du die Parameter als optionale Argumente dazu. Standardwerte sind None

Und dann überprüfst du in den Zeilen jeweils ob die parameter none sind oder nicht. Wenn nicht, dann machs wie bisher, wennse da sind, nutze die. Also quasi
if blabla is None then that else getfunction(die die momentan da steht)